app/src/renderer/vuex/modules/delegation.js

```diff
@@ -139,31 +143,36 @@ export default ({ node }) => {
           })
         }
       }
-
       await dispatch("sendTx", {
         type: "updateDelegations",
         to: rootState.wallet.address, // TODO strange syntax
         delegations: delegate,
         begin_unbondings: unbond
       })
-
       // (optimistic update) we update the atoms of the user before we get the new values from chain
       let atomsDiff = delegations
         // compare old and new delegations and diff against old atoms
         .map(
           delegation =>
-            state.committedDelegates[delegation.delegate.owner] -
-            delegation.atoms
+            calculateTokens(
+              delegation.delegate,
+              state.committedDelegates[delegation.delegate.owner]
+            ) - delegation.atoms
         )
         .reduce((sum, diff) => sum + diff, 0)
       commit("setAtoms", rootState.user.atoms + atomsDiff)
 
       // we optimistically update the committed delegations
-      updateCommittedDelegations(delegations, commit)
+      // updateCommittedDelegations(delegations, commit)
       // TODO usually I would just query the new state through the LCD and update the state with the result, but at this point we still get the old shares
-      dispatch("updateDelegates").then(() =>
-        updateCommittedDelegations(delegations, commit)
-      )
+      setTimeout(async () => {
+        dispatch("updateDelegates") //.then(() =>
```

Comment:
delete comments

Reply:
https://github.com/cosmos/voyager/issues/1300leaving in to be picked up by